### PR TITLE
Feat/price for conversion

### PR DESCRIPTION
**Pull request by Will-Smith11**

### CODE REVIEW

File: crates/angstrom-net/src/peers/manager.rs [hunk 1]
```diff
@@ -102,22 +93,22 @@ impl PeersManager {
...
+//     ///
+//     /// If currently connected this will disconnect the session
+//     Remove(PeerId),
+//     /// Apply a reputation change to the given peer.
+//     ReputationChange(PeerId, ReputationChangeKind),
+//     /// Get information about a peer
+//     GetPeer(PeerId, oneshot::Sender<Option<Peer>>),
+//     /// Get node information on all peers
+//     GetPeers(oneshot::Sender<Vec<NodeRecord>>)
+// }
```
Comment (PlamenHristov) [inline crates/angstrom-net/src/peers/manager.rs:111]: Just delete?

File: crates/angstrom-net/src/peers/manager.rs [hunk 2]
```diff
@@ -161,17 +150,17 @@ enum ReputationChangeOutcome {
...
+    //     Peer { reputation: DEFAULT_REPUTATION, kind, trusted, connected }
+    // }
+    //
+    // /// Resets the reputation of the peer to the default value. This always
+    // /// returns [`ReputationChangeOutcome::None`].
+    // fn reset_reputation(&mut self) -> ReputationChangeOutcome {
+    //     self.reputation = DEFAULT_REPUTATION;
+    //
+    //     ReputationChangeOutcome::None
+    // }
```
Comment (PlamenHristov) [inline crates/angstrom-net/src/peers/manager.rs:163]: Just delete?

File: crates/angstrom-net/src/peers/manager.rs [hunk 3]
```diff
@@ -204,11 +193,11 @@ impl Peer {
...
-    /// Unbans the peer by resetting its reputation
-    #[inline]
-    fn unban(&mut self) {
-        self.reputation = DEFAULT_REPUTATION
-    }
+    // /// Unbans the peer by resetting its reputation
+    // #[inline]
+    // fn unban(&mut self) {
+    //     self.reputation = DEFAULT_REPUTATION
+    // }
```
Comment (PlamenHristov) [inline crates/angstrom-net/src/peers/manager.rs:200]: Just delete?

File: crates/angstrom-net/src/swarm.rs [hunk 4]
```diff
@@ -40,13 +39,14 @@ impl<DB: Unpin> Swarm<DB> {
...
-        }
-    }
+    // pub(crate) fn remove_peer(&mut self, peer_id: PeerId, kind: PeerKind) {
+    //     match kind {
+    //         PeerKind::Basic => self.state.peers_mut().remove_peer(peer_id),
+    //         PeerKind::MevGuard =>
+    // self.state.peers_mut().remove_peer_from_trusted_set(peer_id),
+    //         _ => todo!()
+    //     }
+    // }
```
Comment (PlamenHristov) [inline crates/angstrom-net/src/swarm.rs:49]: Just delete?

File: crates/matching-engine/src/book/order.rs [hunk 5]
```diff
@@ -176,23 +180,23 @@ impl<'a> Order<'a> {
...
+    //             ..lo.clone()
+    //         }),
+    //         Self::AMM(r) => {
+    //             r.fill(filled_quantity);
+    //             // Return a bogus order that we never use
+    //             Self::PartialFill(StandingOrder::default())
+    //         }
+    //     }
+    // }
 }
```
Comment (PlamenHristov) [inline crates/matching-engine/src/book/order.rs:202]: Just delete?

File: crates/types/src/sol_bindings/testnet/random/implementations.rs [hunk 6]
```diff
@@ -85,142 +82,144 @@ impl Distribution<OrderMeta> for Standard {
...
+//             initial_prices:       rng.gen_many(vec_sizes),
+//             pre_transformations:  rng.gen_many_sized::<150>(vec_sizes),
+//             top_of_block_orders:  rng.gen_many(vec_sizes),
+//             swaps:                rng.gen_many(vec_sizes),
+//             orders:               rng.gen_many(vec_sizes),
+//             post_transformations: rng.gen_many_sized::<150>(vec_sizes),
+//             donates:              rng.gen_many(vec_sizes)
+//         }
+//     }
+// }
```
Comment (PlamenHristov) [inline crates/types/src/sol_bindings/testnet/random/implementations.rs:99]: Just delete?

File: crates/types/src/sol_bindings/testnet/random/implementations.rs [hunk 7]
```diff
@@ -85,142 +82,144 @@ impl Distribution<OrderMeta> for Standard {
...
+//             asset0Index:    rng.gen(),
+//             asset1Index:    rng.gen(),
+//             startTick:
+// Signed::from_be_bytes(rng.gen_range(-8388608..=8388607).to_be_bytes()),
+//             totalTicks:     rng.gen(),
+//             startLiquidity: rng.gen(),
+//             amounts0:       rng.gen_many(1)
+//         }
+//     }
+// }
```
Comment (PlamenHristov) [inline crates/types/src/sol_bindings/testnet/random/implementations.rs:222]: Just delete?

File: crates/angstrom-net/src/state.rs [hunk 8]
```diff
@@ -15,14 +15,14 @@ sol! {
 pub struct StromState<DB> {
     peers_manager: PeersManager,
 
-    db:           DB,
+    _db:          DB,
     active_peers: HashSet<PeerId>,
```
Comment (PlamenHristov) [inline crates/angstrom-net/src/state.rs:19]: rm?

File: crates/matching-engine/src/cfmm/uniswap/pool_manager.rs [hunk 9]
```diff
@@ -46,7 +46,7 @@ where
     latest_synced_block: u64,
     state_change_buffer: usize,
     state_change_cache:  Arc<RwLock<StateChangeCache<Loader, A>>>,
-    provider:            Arc<P>,
+    pub provider:        Arc<P>,
     sync_started:        AtomicBool
```
Comment (PlamenHristov) [inline crates/matching-engine/src/cfmm/uniswap/pool_manager.rs:50]: Why touch internals?

Comment (Will-Smith11) [reply crates/matching-engine/src/cfmm/uniswap/pool_manager.rs:50]: needs to be cleaned up, i don't anymore. I found a bug in some of the math i was doing so fixing that now

**Comment by PlamenHristov** [comment]
If I understand correctly you are adding the ability to calc sqrt price. Now I'd be inclined to just add it to the [pool](https://github.com/SorellaLabs/angstrom/blob/0650016449f89aee64b56142f2c67d542eba16a6/crates/matching-engine/src/cfmm/uniswap/pool.rs#L96) using the price calc from [here](https://github.com/darkforestry/amms-rs/blob/791fac345822484bf2bed2b4eca0b55589c20cad/src/amm/uniswap_v3/mod.rs#L132C4-L146C10). 

Then since you already have the `SyncedUniswapPools` you can get it directly from the pool?

**Comment by Will-Smith11** [comment]
> If I understand correctly you are adding the ability to calc sqrt price. Now I'd be inclined to just add it to the [pool.rs](https://github.com/SorellaLabs/angstrom/blob/0650016449f89aee64b56142f2c67d542eba16a6/crates/matching-engine/src/cfmm/uniswap/pool.rs#L96) using the price calc from [here](https://github.com/darkforestry/amms-rs/blob/791fac345822484bf2bed2b4eca0b55589c20cad/src/amm/uniswap_v3/mod.rs#L132C4-L146C10).
> 
> Then since you already have the `SyncedUniswapPools` you can get it directly from the pool?
> 
> (I assume we are only interested in latest prices)

so that function is decimal adjusted. I don't want that, and then the other math im doing isn't related to this

**Pull request merged**